packages/core/src/external/carequality/command/documents/send-ihe-gateway-results.ts

```diff
@@ -1,22 +1,18 @@
 import { Sequelize, QueryTypes } from "sequelize";
-import axios from "axios";
 import dayjs from "dayjs";
 import duration from "dayjs/plugin/duration";
 import { RaceControl, checkIfRaceIsComplete, controlDuration } from "../../../../util/race-control";
 import { MetriportError } from "../../../../util/error/metriport-error";
 import { initSequelizeForLambda } from "../../../../util/sequelize";
 import { errorToString } from "../../../../util/error/shared";
-
-const REQUEST_ID_COLUMN = "request_id";
-
-const api = axios.create();
+import { REQUEST_ID_COLUMN } from "../../ihe-result";
 
 dayjs.extend(duration);
 
-export const CONTROL_TIMEOUT = dayjs.duration({ minutes: 3 });
-const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 5 });
+export const CONTROL_TIMEOUT = dayjs.duration({ minutes: 15 });
+const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 30 });
 
-export const sendIHEGatewayResults = async <TableResult>({
+export const pollIHEGatewayResults = async <TableResult>({
```

Comment:
use `function` for function declaration - [reference](https://www.notion.so/metriport/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?pvs=4#b669ec3473f1414b9d234cbdac20041b)

Reply:
re: generics
This is not how we should do Generics. The goal is not to pass a type from the caller with no constraints on the actual function, this is just bypassing the compiler.
We should have a type on the place that builds the object.
In this case, [this line](https://github.com/metriport/metriport/pull/1437/files/7ffc6ec89c86d810dfbe66a71f70c28abb519abb..2b99e07393ee12336aec40ffb5acbf4aaed7e6fd#diff-6d599c1b0e788cf54be20015bba38265d09d8c2403265048de7a0805761bcf9aR111) is just using `as` to the type we get here as a parameter. We could pass `Car`, `Dog` or whatever and the compiler would not be able to save us from the runtime error.